csrc/device_lower/analysis/index_compute.cpp

```diff
@@ -421,6 +451,20 @@ IndexingParameters getPredicateInitialIndexParameters(
         loop_to_ind_map[loop] = SimplifyingIrBuilder::subExpr(
             loop_id->extent(), GpuLower::current()->kernel()->oneVal());
       }
+
+      // When predicating a loop at the maximum end, predicate
+      // expressions such as (extent-1) are used, which represent the
+      // maximum possible value of the loop range but are not
+      // guaranteed to result in the maximum index when traversing
+      // through merge inner domains as modulo is used. Keep track of
+      // those domains, which will be used by IndexCompute to make
+      // necessary adjustments. See also csrc/index_compute.h for more
+      // context.
+      if (!is_start_predicate && trackUnswitchedDomain(loop)) {
+        index_parameters.unswitched_domains.insert(
+            GpuLower::current()->caMap()->getConcreteMappedID(
+                loop_id, IdMappingMode::EXACT));
```

Comment:
Not necessarily. We don't want to track vectorized domains or domains mapped with vectorized domains. They should still need to be filtered out even if this code is moved there.

Also, tracking may be necessary even if the loop is parallelized when the domain is extended for halo. I haven't tested such patterns, but placing this tracking here seems to make most sense to me.

Reply:
On a slightly related note, I'm not sure why I did this way:
 https://github.com/NVIDIA/Fuser/pull/687/files/88183264b5ad608e6efd3af8adb22573eb269937#diff-2d7e9c403d33ef6dd38c80bfa051f541416532aabe7679112e7d3ca94c51f6e9R443-R444

It seems it should be `loop_to_ind_map[loop] = loop->stop() - 1`

Reply:
I think we don't have to handle IDs mapped to vectorization explicitly, because it will be automatically handled by the `inner_extent % total_extent == 0` check in `isModuloInvalidUnswitchedIndex`, but still, let's just leave this PR as is and handle it explicitly, because why not?

Reply:
I remember the check was necessary to avoid some predicate changes. Let me check again.

Reply:
I think it was just because `inner_extent` in `isModuloInvalidUnswitchedIndex` may not be statically known, so 

> because it will be automatically handled by the inner_extent % total_extent == 0 check in isModuloInvalidUnswitchedIndex

may not hold.